targets/XPlatCppSdk/source/code/source/playfab/PlayFabAndroidHttpPlugin.cpp

```diff
@@ -0,0 +1,743 @@
+        {
+
+        }
+
+        virtual ~PlayFabAndroidHelper()
+        {
+            if (httpRequestClass)
+            {
+                JNIEnv* jniEnv;
+                if(GetJniEnv(&jniEnv))
+                {
+                    jniEnv->DeleteGlobalRef(httpRequestClass);
+                }
+                httpRequestClass = nullptr;
+            }
+            this->jvm = nullptr;
+        }
+
+        bool Initialize(JavaVM* jvm)
+        {
+            assert(jvm);
+            this->jvm = jvm;
+
+            JNIEnv* jniEnv;
+            this->jvm->GetEnv(reinterpret_cast<void**>(&jniEnv), JNI_VERSION_1_6);
+            assert(jniEnv);
+            if (jniEnv == nullptr)
+            {
+                return false;
+            }
+
+            jclass localClass = jniEnv->FindClass("com/microsoft/xplatcppsdk/HttpRequest");
+            assert(localClass);
+            if (localClass == nullptr)
+            {
+                return false;
+            }
+
+            httpRequestClass = reinterpret_cast<jclass>(jniEnv->NewGlobalRef(localClass));
+            assert(httpRequestClass);
+
+            jniEnv->DeleteLocalRef(localClass);
+
+            if (httpRequestClass == nullptr)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        bool GetJavaVM(JavaVM** jvm)
+        {
+            assert(this->jvm);
+            if (this->jvm == nullptr)
+            {
+                assert(false);
+                return false;
+            }
+
+            *jvm = this->jvm;
+
+            return true;
+        }
+
+        bool GetJniEnv(JNIEnv** jniEnv)
+        {
+            assert(this->jvm);
+            if (this->jvm == nullptr)
+            {
+                assert(false);
+                return false;
+            }
+
+            jint result = this->jvm->GetEnv(reinterpret_cast<void**>(jniEnv), JNI_VERSION_1_6);
+
+            if (result != JNI_OK)
+            {
+                assert(false);
+                return false ;
+            }
+
+            return true;
+        }
+
+        jclass GetHttpRequestClass()
+        {
+            return httpRequestClass;
+        }
+
+    private:
+        JavaVM* jvm;
+        jclass httpRequestClass;
+    };
+
+}
+
+static std::unique_ptr<PlayFab::PlayFabAndroidHelper> s_helper;
+
+static PlayFab::PlayFabAndroidHelper& GetHelper()
+{
+    assert(s_helper);
+    return *s_helper;
+}
+
+JNIEXPORT jint JNI_OnLoad(JavaVM* vm, void* reserved)
+{
+    __android_log_print(ANDROID_LOG_INFO, "XPlatCppSdk", "JNI_OnLoad");
+
+    try
+    {
+        s_helper = std::make_unique<PlayFab::PlayFabAndroidHelper>();
+    }
+    catch (std::exception& e)
+    {
+        __android_log_print(ANDROID_LOG_INFO, "XPlatCppSdk", "Failed to create PlayFabAndroidHelper");
+        throw e;
+    }
+    if (!(s_helper->Initialize(vm)))
+    {
+        throw "Initializing to load classes failed";
+    }
+
+    return JNI_VERSION_1_6;
```

Comment:
>JNI_VERSION_1_6 [](start = 11, length = 15)

Does this need to be changed for different version of NDK / Android OS versions ?

Reply:
This time, Android uses 1.6 version. And this is for just let JNI system to know how this JNI plug-in should be handled. So don't need to change this version until we uses new version architectre. And new version of JNI system will be load using 1.6 interface.

---
In reply to: [273701150](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r273701150) [](ancestors = 273701150)